apps/web/src/components/NotificationList/NotificationItem.tsx

```diff
@@ -104,6 +136,11 @@ const NotificationItem = ({
   );
   const [{ fetching: markingAsUnread }, executeMarkAsUnreadMutation] =
     useMutation(MarkNotificationAsUnread_Mutation);
+  const [{ fetching: downloadingFile }, executeFileDownload] =
```

Comment:
Seems a bit odd to me that the file download hook and other related logic is here whether this is a file-download-type notification or not. Maybe it would make more sense to have two different components, NotificationItem and  DownloadNotificationItem? Maybe DownloadNotificationitem could wrap the other one and add some extra logic?
Maybe there's no way of avoiding mostly-duplicated content and its not worth it - just a thought!

Reply:
Yeah, that is a good suggestion. Honestly, I was not really happy with this solution either so I'll spend more time seeing if we can solve it with a better solution building off your suggestion. :sweat_smile:

Reply:
Refactored in [refactor download link into own component](https://github.com/GCTC-NTGC/gc-digital-talent/pull/10992/commits/3cfa6fcd67dd67ce21786c0dfb5af8c9cffdc2ae)